testing/http_authentication.rst

```diff
@@ -54,3 +52,71 @@ key in your firewall, along with the ``form_login`` key:
                 ),
             ),
         ));
+
+Tests can now authenticate via HTTP passing the username and password as server
+variables using the second argument of ``createClient()``::
+
+    $client = static::createClient(array(), array(
+        'PHP_AUTH_USER' => 'username',
+        'PHP_AUTH_PW'   => 'pa$$word',
+    ));
+
+The username and password can also be passed on a per request basis::
+
+    $client->request('DELETE', '/post/12', array(), array(), array(
+        'PHP_AUTH_USER' => 'username',
+        'PHP_AUTH_PW'   => 'pa$$word',
+    ));
+
+Creating the Authentication Token
+---------------------------------
+
+If your application uses a more advanced authentication mechanism, you can't
+use the previous trick, but it's still possible to make tests faster. The trick
+now is to bypass the authentication process, create the *authentication token*
+yourself and store it in the session.
+
+This technique requires some knowledge of the security component internals,
+but the following example shows a complete example that you can adapt to your
+needs::
+
+    // src/AppBundle/Tests/Controller/DefaultControllerTest.php
+    namespace Appbundle\Tests\Controller;
+
+    use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+    use Symfony\Component\BrowserKit\Cookie;
+    use Symfony\Component\Security\Core\Authentication\Token\UsernamePasswordToken;
+
+    class DefaultControllerTest extends WebTestCase
+    {
+        private $client = null;
+
+        public function setUp()
+        {
+            $this->client = static::createClient();
+        }
+
+        public function testSecuredHello()
+        {
+            $this->logIn();
+            $crawler = $this->client->request('GET', '/admin');
+
+            $this->assertTrue($this->client->getResponse()->isSuccessful());
```

Comment:
Please make use of `$this->assertSame(Response::HTTP_OK, $this->client->getResponse());` which is much easier to debug, providing the wrong status code instead of just `false` when failing.